django/contrib/auth/views.py

```
@@ -103,10 +103,13 @@ def get_form_kwargs(self):
         kwargs["request"] = self.request
         return kwargs
 
-    def form_valid(self, form):
-        """Security check complete. Log the user in."""
-        auth_login(self.request, form.get_user())
+        user = form.get_user()
```

Ruff (0.12.2)

Undefined name `form`

(F821)